### PR TITLE
Add intersection method to Range class

### LINKED ISSUE #9509: Python postgres `Range` impl missed `intersection` method
### Describe the use case

Python postgres `Range` impl missed `intersection` method:

```sql
SELECT '[1, 10]'::int4range * '[5, 15]'::int4range  -- [5, 11)
```

### Databases / Backends / Drivers targeted

postgresql

### Example Use

```py
a = Range(1, 10)
b = Range(5, 15)

c = a * b
```

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by uriyyo**
### Description
Fixes: #9509 

### Checklist

This pull request is:

- [ ] A documentation / typographical error fix
	- Good to go, no issue or tests are needed
- [ ] A short code fix
	- please include the issue number, and create an issue if none exists, which
	  must include a complete example of the issue.  one line code fixes without an
	  issue and demonstration will not be accepted.
	- Please include: `Fixes: #<issue number>` in the commit message
	- please include tests.   one line code fixes without tests will not be accepted.
- [x] A new feature implementation
	- please include the issue number, and create an issue if none exists, which must
	  include a complete example of how the feature would look.
	- Please include: `Fixes: #<issue number>` in the commit message
	- please include tests.

**Have a nice day!**

### CODE REVIEW

File: doc/build/changelog/unreleased_20/9509.rst [hunk 1]
```diff
@@ -0,0 +1,6 @@
+.. change::
+      :tags: usecase, postgresql
+      :tickets: 9509
+
+      Add missed :meth:`_postgresql.Range.intersection` method.
```
Comment (CaselIT) [inline doc/build/changelog/unreleased_20/9509.rst:5]: Missing

**Review by CaselIT — COMMENTED** [review]
Thanks

File: doc/build/changelog/unreleased_20/9509.rst (hunk 1 above)
Comment (uriyyo) [reply doc/build/changelog/unreleased_20/9509.rst]: Thanks, fixed

**Review by sqla-tester — COMMENTED** [review]
OK, this is **sqla-tester** setting up my work on behalf of **CaselIT** to try to get revision 45856974871d846d5cfb7e44aae10c2c3c682ac6 of this pull request into gerrit so we can run tests and reviews and stuff

**Comment by sqla-tester** [comment]
New Gerrit review created for change 45856974871d846d5cfb7e44aae10c2c3c682ac6: https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4523

**Review by sqla-tester — COMMENTED** [review]
OK, this is **sqla-tester** setting up my work on behalf of **CaselIT** to try to get revision 8c663dbfe95be4680afecc4cd8003cf0260584b2 of this pull request into gerrit so we can run tests and reviews and stuff

**Comment by sqla-tester** [comment]
Patchset 8c663dbfe95be4680afecc4cd8003cf0260584b2 added to existing Gerrit review https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4523

File: test/dialect/postgresql/test_types.py [hunk 2]
```diff
@@ -4393,6 +4393,50 @@ def test_difference(self, connection, r1t, r2t):
...
+        q = select(
+            cast(r1, RANGE).intersection(r2),
+        )
+        validate_q = select(
+            literal_column(f"'{r1}'::{range_typ}*'{r2}'::{range_typ}", RANGE),
+        )
+
+        pg_res = connection.execute(q).scalar()
+
+        validate_difference = connection.execute(validate_q).scalar()
```
[the remark this answers is not in the document]
Comment (uriyyo) [reply test/dialect/postgresql/test_types.py]: Sorry, my bad. Fixed, now it should work as expected.

**Comment by CaselIT** [comment]
Rebase or merge should be the same. We can simply rebase form gerrit in case

**Review by sqla-tester — COMMENTED** [review]
OK, this is **sqla-tester** setting up my work on behalf of **CaselIT** to try to get revision 596648e7989327eef1807057519b2295b48f1adf of this pull request into gerrit so we can run tests and reviews and stuff

**Comment by sqla-tester** [comment]
Patchset 596648e7989327eef1807057519b2295b48f1adf added to existing Gerrit review https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4523

**Comment by lelit** [comment]
AFAICT the failure is not related to these changes, is it?

**Comment by uriyyo** [comment]
Agree with @lelit, the failure is not related to changes made in PR.

**Review by sqla-tester — COMMENTED** [review]
OK, this is **sqla-tester** setting up my work on behalf of **CaselIT** to try to get revision 596648e7989327eef1807057519b2295b48f1adf of this pull request into gerrit so we can run tests and reviews and stuff

**Comment by uriyyo** [comment]
@CaselIT CI passed, I guess now you can review this PR.

**Comment by CaselIT** [comment]
sure, I've approved it. Waiting for Mike to take a look.

Thanks!

**Comment by sqla-tester** [comment]
**mike bayer** (zzzeek) wrote:

this is great, thanks for the efforts!

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4523

**Comment by sqla-tester** [comment]
Gerrit review https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4523 has been **merged**. Congratulations! :)

**Pull request closed without merging**